### PR TITLE
Reland: Create PlatformView instance right after method channel call from Dart

### PR DESCRIPTION
**Pull request by blasten**
Relands https://github.com/flutter/engine/pull/20500

1f15b7e fixes the issue.

### CODE REVIEW

File: shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java [hunk 1]
```diff
@@ -146,12 +158,17 @@ public void createAndroidViewForPlatformView(
         public void disposeAndroidViewForPlatformView(int viewId) {
           // Hybrid view.
           final View platformView = platformViews.get(viewId);
+          final FlutterMutatorView parentView = platformViewParent.get(viewId);
           if (platformView != null) {
-            final FlutterMutatorView mutatorView = mutatorViews.get(viewId);
-            mutatorView.removeView(platformView);
-            ((FlutterView) flutterView).removeView(mutatorView);
+            if (parentView != null) {
```
Comment (gaaclarke) [inline shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java:163]: This is a duplicate conditional

File: shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java [hunk 2]
```diff
@@ -792,22 +810,20 @@ private void finishFrame(boolean isFrameRenderedUsingImageReaders) {
       }
     }
 
-    for (int i = 0; i < platformViews.size(); i++) {
-      int viewId = platformViews.keyAt(i);
-      View platformView = platformViews.get(viewId);
-      View mutatorView = mutatorViews.get(viewId);
+    for (int i = 0; i < platformViewParent.size(); i++) {
+      final int viewId = platformViewParent.keyAt(i);
+      final View parentView = platformViewParent.get(viewId);
```
Comment (gaaclarke) [inline shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java:815]: Is it possible parentView will be null?

Comment (blasten) [reply shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java:815]: It's not possible. This is iterating over the existing entries, and the code never puts `null` in the dictionary.

File: shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java [hunk 3]
```diff
@@ -79,9 +79,20 @@
   // it is associated with(e.g if a platform view creates other views in the same virtual display.
   private final HashMap<Context, View> contextToPlatformView;
 
-  private final SparseArray<PlatformViewsChannel.PlatformViewCreationRequest> platformViewRequests;
+  // The view returned by `PlatformView#getView()`.
```
Comment (cyanglaz) [inline shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java:82]: nits: views

Comment (blasten) [reply shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java]: Done

**Pull request merged**